src/components/forms/uncontrolled-form/autocomplete-input.tsx

```diff
@@ -0,0 +1,41 @@
+import type { FC, ReactNode } from "react";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { classNames } from "@/lib/class-names";
+
+interface AutocompleteInputProps {
+  error?: string;
+  id: string;
+  label: ReactNode | string;
+  name: string;
+  options: string[];
+}
+
+export const AutocompleteInput: FC<AutocompleteInputProps> = ({
+  error,
+  id,
+  label,
+  name,
+  options,
+}) => {
+  return (
+    <div className="relative mb-4">
+      <Label htmlFor={id}>{label}</Label>
+      <Input id={id} list={`${id}-list`} name={name} />
+      <datalist id={`${id}-list`}>
+        {options.map((c) => (
+          <option key={c} value={c} />
+        ))}
```

Comment:
Good use of `key` for each option. Consider renaming `c` to something more descriptive like `option` for clarity.
```suggestion
        {options.map((option) => (
          <option key={option} value={option} />
        ))}
```